src/components/Card.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useEffect, FC } from "react";
+import { IPostInterface } from "components/models";
+import PostComponent from "containers/PostCardContainer";
+import {
+  FieldWrapper,
+  CardWrapper,
+  InputWrapper,
+  CrossButtonWrapper,
+  StyledButton,
+} from "style";
+
+interface ICardInterface {
+  title: string;
+  posts: Array<IPostInterface>;
+  getData: () => void;
+  postData: (obj: string) => void;
+  closePressed: () => void;
+}
+
+const Card: FC<ICardInterface> = ({
+  title,
+  posts,
+  getData,
+  postData,
+  closePressed,
+}) => {
+  const [data, setData] = useState({
+    showForm: false,
+    text: "",
+    title: "",
+    location: "",
+    isEdit: false,
+  });
+
+  useEffect(() => {
+    getData();
+  }, [getData]);
+
+  function addItem() {
+    postData(data.text);
+    setData({ ...data, showForm: false, text: "" });
+  }
+
+  function showForm() {
+    setData({ ...data, showForm: true, text: "" });
+    closePressed();
+  }
+
+  return (
+    <CardWrapper>
+      <h6>{title}</h6>
+      <div>
+        <div>
+          <InputWrapper
+            onChange={(evt: any) =>
+              setData({ ...data, title: evt.target.value })
+            }
+            value={data.title}
+            placeholder="Title"
+          />
+
+          <InputWrapper
+            onChange={(evt: any) =>
+              setData({ ...data, location: evt.target.value })
+            }
+            value={data.location}
+            placeholder="Location"
+          />
+          <InputWrapper
+            onChange={(evt: any) =>
+              setData({ ...data, text: evt.target.value })
+            }
+            value={data.text}
```

Comment:
use formik here instead of states variables.

Also take this form out of this file and create a separate file for this.